test/bridges/base-bridge-receiver-test.ts

```diff
@@ -0,0 +1,452 @@
+    const {
+      baseBridgeReceiver,
+      govTimelock,
+      localTimelock
+    } = await makeBridgeReceiver();
+
+    expect(await baseBridgeReceiver.proposalCount()).to.eq(0);
+
+    const delay = await localTimelock.delay();
+    const newDelay = delay.mul(2);
+
+    const calldata = encodeBridgeReceiverCalldata({
+      targets: Array(1).fill(localTimelock.address),
+      values: Array(1).fill(0),
+      signatures: Array(1).fill('setDelay(uint256)'),
+      calldatas: [
+        utils.defaultAbiCoder.encode(['uint256'], [newDelay.toNumber()])
+      ]
+    });
+
+    await baseBridgeReceiver.processMessageExternal(govTimelock.address, calldata);
+
+    // queues the proposal
+    expect(await baseBridgeReceiver.state(1)).to.eq(ProposalState.Queued);
+
+    const { eta } = await baseBridgeReceiver.proposals(1);
+
+    await ethers.provider.send('evm_setNextBlockTimestamp', [eta.toNumber()]);
+
+    const tx = await wait(baseBridgeReceiver.executeProposal(1));
+
+    expect(await localTimelock.delay()).to.eq(newDelay);
+
+    expect(event(tx, 2)).to.be.deep.equal({ ProposalExecuted: { id: 1n } });
+
+    const updatedProposal = await baseBridgeReceiver.proposals(1);
+
+    expect(updatedProposal.executed).to.be.true;
+    expect(await baseBridgeReceiver.state(1)).to.eq(ProposalState.Executed);
+  });
+
+  it('executeProposal > reverts if proposal is already executed', async () => {
+    const {
+      baseBridgeReceiver,
+      govTimelock,
+      localTimelock
+    } = await makeBridgeReceiver();
+
+    expect(await baseBridgeReceiver.proposalCount()).to.eq(0);
+
+    const calldata = encodeBridgeReceiverCalldata({
+      targets: Array(1).fill(localTimelock.address),
+      values: Array(1).fill(0),
+      signatures: Array(1).fill('setDelay(uint256)'),
+      calldatas: [
+        utils.defaultAbiCoder.encode(['uint256'], [20 * 60])
+      ]
+    });
+
+    await baseBridgeReceiver.processMessageExternal(govTimelock.address, calldata);
+
+    const { eta } = await baseBridgeReceiver.proposals(1);
+
+    await ethers.provider.send('evm_setNextBlockTimestamp', [eta.toNumber()]);
+
+    await baseBridgeReceiver.executeProposal(1);
+
+    await expect(
+      baseBridgeReceiver.executeProposal(1)
+    ).to.be.revertedWith("custom error 'ProposalNotQueued()'");
+  });
+
+  it('state > reverts for proposal id = 0', async () => {
+    const { baseBridgeReceiver } = await makeBridgeReceiver();
+
+    await expect(
+      baseBridgeReceiver.executeProposal(0)
+    ).to.be.revertedWith("custom error 'InvalidProposalId()'");
+  });
+
+  it('state > reverts if proposal id is greater than proposalCount', async () => {
+    const { baseBridgeReceiver } = await makeBridgeReceiver();
+
+    const proposalCount = await baseBridgeReceiver.proposalCount();
+
+    await expect(
+      baseBridgeReceiver.executeProposal(proposalCount.add(1))
+    ).to.be.revertedWith("custom error 'InvalidProposalId()'");
+  });
+});
```

Comment:
We should also test the case where a transaction is already queued.

Reply:
Is that covered by: `processMessage > reverts for repeated transactions`?

Reply:
Ah I missed it. Nice work!